airflow/utils.py

```diff
@@ -617,11 +619,20 @@ def handle_timeout(self, signum, frame):
         raise AirflowTaskTimeout(self.error_message)
 
     def __enter__(self):
-        signal.signal(signal.SIGALRM, self.handle_timeout)
-        signal.alarm(self.seconds)
+        try:
+            signal.signal(signal.SIGALRM, self.handle_timeout)
+            signal.alarm(self.seconds)
+        except Exception as e:
```

Comment:
Might be a good idea to log the exception (and maybe even the trace) too, even if it's only debug level